lib/mydevice/mydevice.py

```diff
@@ -79,3 +79,29 @@ def back_button(self):
         :return: none
         """
         self._device.press.back()
+
+    def call_uiautomator(self,number):
+        """
+        It receives a telephone number and makes the call with uiautomator
+        Pre-condition: the number has to be valid and the phone application has to be on the home page
+        param str numner: telephone number
+        returns: 
+        """
+        self._device.screen.on()
+        self._device.press.home()
+        self._device(text="Phone").click()
+        self._device(resourceId="com.android.dialer:id/floating_action_button").click()
+        arr_num = list(number)
+        for digito in arr_num:
```

Comment:
puedes validar que el formato del número de teléfono sea el correcto, así como su longitud.

Reply:
```suggestion
        Phone_number = ''.join(i for i in number if i.isdigit())
        if len(Phone_number) != 10:
            print("Formato incorrecto, tiene que tener un tamaño de 10 digitos: 4491234567")
            return False
```

Con este codigo, primero tomo la cadena que recibe como numero, y la convierto en puros digitos en dado caso que se ingrese con guiones, espacios, etc... y despues valido que el tamaño de la cadena sea de 10 digitos, sino es de ese tamaño regresa un false. Esta correcto eso?

Reply:
Sería mejor levantar una excepción donde menciones que el formato es incorrecto.

`raise ValueError("Invalid phone number format {}".format(number))`